modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/Select/Select.es.js

```diff
@@ -542,4 +542,63 @@ describe('Select', () => {
 			'item11',
 		]);
 	});
+
+	it('shows the options value if there are values', async () => {
+		const handleFieldEdited = jest.fn();
+
+		render(
+			<SelectWithProvider
+				dataSourceType="manual"
+				multiple={true}
+				onChange={handleFieldEdited}
+				options={createOptions(12)}
+				predefinedValue={['item1', 'item2']}
+				spritemap={spritemap}
+				value={['item3']}
+			/>
+		);
+
+		expect(document.querySelector('span[value="item1"]')).toBeNull();
+		expect(document.querySelector('span[value="item2"]')).toBeNull();
+		expect(document.querySelector('span[value="item3"]')).not.toBeNull();
+	});
+
+	it('shows the predefinedValues if there is no value', async () => {
+		const handleFieldEdited = jest.fn();
+
+		render(
+			<SelectWithProvider
+				dataSourceType="manual"
+				multiple={true}
+				onChange={handleFieldEdited}
+				options={createOptions(12)}
+				predefinedValue={['item1', 'item2']}
+				spritemap={spritemap}
+				value={[]}
+			/>
+		);
+
+		expect(document.querySelector('span[value="item1"]')).not.toBeNull();
+		expect(document.querySelector('span[value="item2"]')).not.toBeNull();
```

Comment:
- Prefer use [testing library queries](https://testing-library.com/docs/queries/about/), because they enforce the code accessibility 
- Prefer assertions methods from [jest-dom](https://github.com/testing-library/jest-dom), because they provide a more accurate error message in case of failure
```suggestion
		expect(getByText('item1')).toBeInTheDocument();
		expect(getByText('item2')).toBeInTheDocument();
```
**Note:**
- depends on the component's implementation you won't be able to use testing library queries

Reply:
The tests related to Select field didn't work with the suggestion. I've tried other ways and didn't work too. So, I will keep it as it is to not take so long.

Reply:
I'm ok with the use of the `querySelector`, but I think you are still able use jest-dom assertions with no problem in this case:
```suggestion
		expect(document.querySelector('span[value="item1"]')).toBeInTheDocument();
		expect(document.querySelector('span[value="item2"]')).toBeInTheDocument();
```